services/web/server/src/simcore_service_webserver/projects/_projects_service.py

```diff
@@ -1603,9 +1587,31 @@ async def close_project_for_user(
         # remove the project from our list of opened ones
         await user_session.remove(key=PROJECT_ID_KEY)
 
+        # remove the clent session from the project room
+        _socket_id = await user_session.get_socket_id()
+        if _socket_id is not None:
+            _logger.debug(
+                "User %s/%s is leaving project room %s with socket_id %s",
+                user_id,
+                client_session_id,
+                project_uuid,
+                _socket_id,
+            )
+            sio = get_socket_server(app)
+            await sio.leave_room(
+                _socket_id, SocketIORoomStr.from_project_id(project_uuid)
+            )
+        else:
+            _logger.warning(
```

Comment:
I believe this is unexpected behavior right?
1. nobody is going to check these warnings, so it's the same as not putting anything. 
2. make it at least an error as this is unexpected behavior.
3. to make it more readable, maybe create a `def leave_room` wrapper that does this. This module is already horrible (I am working on it as well)